### DISCUSSION TITLE
4차 기능 명세서

### DISCUSSION THREAD

**Discussion by kyum-q** [woowacourse-teams/2024-code-zap#360] [기획 관련]
# 4차 기능 명세서

## 스프린트 요구사항

- 프로덕션 서버에 서비스 띄우기
- 실 서버 도메인 연결, HTTPS 적용
- DB 데이터를 drop 하지 않는다
   - DB 계정의 drop 명령어를 제한하라는 의미가 아니다.
   - 실제 운영 중인 서비스는 사용자의 데이터가 담긴 테이블을 drop 하지 않는다.
   - 실 서비스처럼 DB를 운용하라는 의도

## 추가 기능 구현

- 템플릿 목록에서 템플릿 선택 삭제
- 템플릿 목록에서 코드 썸네일 추가
  - 코드 썸네일을 위에 10줄을 기본
  - ~~보여주고 싶은 부분을 선택할 수 있다~~
- 카테고리 수정/삭제 구현
- 회원 정보 수정 + 조회 기능
   - 비밀번호 찾기, 닉네임 수정

## 이 외의 기능
- 사용자 경험을 받아서 리펙터링

**Comment by kyum-q** [comment]
## 아이디어 - 관리 측면 (만두)

**Reply by kyum-q** [reply]
- 버전 관리

**Reply by kyum-q** [reply]
- 템플릿 목록에서 템플릿 선택 삭제

**Reply by kyum-q** [reply]
- 즐겨찾기

**Reply by kyum-q** [reply]
- 템플릿 목록에서 코드 썸네일 추가 
  - 코드 썸네일을 위에 10줄을 기본
  - 보여주고 싶은 부분을 선택할 수 있다

**Reply by kyum-q** [reply]
- 카테고리 수정 페이지 구현

**Reply by kyum-q** [reply]
- 검색 기능 속도 개선

**Comment by kyum-q** [comment]
## 아이디어 - 검색 측면 (머루)

> 현재 해당 기능은 하지 않는다

**Reply by zangsu** [reply]
다른 사람 목록 조회

**Reply by zangsu** [reply]
모든 템플릿 목록 조회

**Reply by zangsu** [reply]
전체 템플릿에서 검색 기능 제공

**Comment by kyum-q** [comment]
# 사용자 시나리오

**Reply by kyum-q** [reply]
## 🥟 만두 (예비 개발자 사용자)

**목표**: 우테코 최종 코딩테스트에 합격하기 위해 코드 스니펫을 빠르게 찾고 자주 쓰는 코드 템플릿을 준비하기

**시나리오**:
- 만두는 우테코 코딩테스트를 준비하기 위해 '코드잽'을 실행합니다.
- 만두는 몇 번째 프리코스에서 구현했는지에 따라 카테고리를 구분하였습니다.
   - 카테고리 [로또, 자동차경주, 크리스마스] 
- 만두는 우테코 동안 구현한 코드 템플릿을 작성합니다.
- 만두는 '자동차 경주' 미션 때 구현한 'JWT를 이용한 로그인' 코드 스니펫을 신속하게 찾고자 합니다.
- 하지만 열정적인 만두는 너무 많은 템플릿을 만들어놔서 찾기 어려웠습니다.
- 만두는 '자동차 경주' 카테고리에서 '로그인'으로 검색을 합니다.
- 하지만 만두는 템플릿의 이름을 'JWT이용한 Login 구현'으로 생성해놔서 검색 결과로 JWT가 없었습니다.
- 그리고 포괄적인 이름의 템플릿인 경우 JWT를 이용했는지 템플릿 상세 페이지를 확인해야 했습니다.
- 만두는 검색을 편리하게 하기 위해 템플릿에 'JWT', 'Login' 태그를 추가했습니다.
- 만두는 시간이 지나 우테코 최종 코딩 테스트를 보러왔습니다.
- 코딩 테스트에서 JWT를 이용한 로그인 기능 구현이 필수 구현 항목에 있었습니다.
- 만두는 '코드잽'을 실행합니다.
- 만두는 '자동차 경주' 카테고리로 들어갑니다. 
- 태그 목록에서 'JWT', 'Login'를 선택해 JWT 로그인과 관련된 코드템플릿만 확인합니다.
- 그리고 태그 목록과 제목 등을 통해 JWT의 로그인 기능을 빠르게 찾습니다.
- 만두는 한번의 클릭으로 쉽게 코드를 재사용합니다.
- 만두는 원하는 기능의 코드를 빠르게 찾아 시간을 대폭 단축하여 우테코 최종 코딩테스트에 합격합니다.

**Comment by zangsu** [comment]
+) 카테고리 수정 페이지 / 회원 정보 수정 + 조회 기능 (비밀번호 찾기, 닉네임 수정...) 등등이 더 필요할 것 같아요~~

**Comment by zangsu** [comment]
4차: 프로덕션(prod) 환경 구축
프로덕션 서버에 서비스 띄우기
실 서버 도메인 연결, HTTPS 적용
DB 데이터를 drop 하지 않는다.
DB 계정의 drop 명령어를 제한하라는 의미가 아니다.
실제 운영 중인 서비스는 사용자의 데이터가 담긴 테이블을 drop 하지 않는다.
실 서비스처럼 DB를 운용하라는 의도

**Comment by kyum-q** [comment]
## 도메인

**Reply by kyum-q** [reply]
### 켬미

- 템플릿 목록에서 템플릿 선택 삭제
- 템플릿 목록에서 코드 썸네일 추가
  -  코드 썸네일을 위에 10줄을 기본
  - 보여주고 싶은 부분을 선택할 수 있다
  
### 몰리
  
- 카테고리 수정/삭제 구현
- 회원 정보 수정 + 조회 기능
  - 비밀번호 찾기, 닉네임 수정

**Reply by kyum-q** [reply]
### Swagger

> 모든 예외처리 관련한 것들은 swagger에서 볼 수 있도록

- 개발 전에 swagger 먼저 구현
